src/video/types.rs

```diff
@@ -122,6 +122,31 @@ impl VideoEncodeType {
             _ => VideoEncodeType::Unknown(fourcc),
         }
     }
+
+    pub fn to_codec(self) -> String {
+        match self {
+            VideoEncodeType::H264 => "video/x-h264",
+            VideoEncodeType::H265 => "video/x-h265",
+            // TODO: We need to handle the mpeg version one day, but not today
+            VideoEncodeType::Mjpg => "video/mpeg",
+            VideoEncodeType::Yuyv => "video/x-raw,format=I420",
+            VideoEncodeType::Unknown(codec) => {
+                return codec;
+            }
+        }
+        .to_string()
+    }
+
+    pub fn from_codec(codec: &str) -> VideoEncodeType {
+        match codec {
+            "video/x-h264" => VideoEncodeType::H264,
+            "video/x-h265" => VideoEncodeType::H265,
+            // TODO: We need to handle the mpeg version one day, but not today
+            "video/mpeg" => VideoEncodeType::Mjpg,
+            "video/x-raw,format=I420" => VideoEncodeType::Yuyv,
+            codec => VideoEncodeType::Unknown(codec.to_string()),
+        }
+    }
```

Comment:
I didn't test it, but Maybe it's possible to just implement From? Something like:

```Rust
impl From<&str> for VideoEncodeType {
    fn from(codec: &str) -> Self {
        match codec {
            "video/x-h264" => VideoEncodeType::H264,
            "video/x-h265" => VideoEncodeType::H265,
            "video/mpeg" => VideoEncodeType::Mjpg,
            "video/x-raw,format=I420" => VideoEncodeType::Yuyv,
            codec => VideoEncodeType::Unknown(codec.to_string()),
        }
    }
}
```

Reply:
Not 100% sure, that's why I created from_codec. The from string would be more generic also including "h264", "h264" and etc.
I would prefer creating a from string that is more generic than from a codec string. 
@joaoantoniocardoso what you think ?
```rs
impl From<&str> for VideoEncodeType {
    fn from(input: &str) -> Self {
        match input.to_lowercase() {
            "h264" => VideoEncodeType::H264,
            "h265" => VideoEncodeType::H265,
            "mpeg" => VideoEncodeType::Mjpg,
            "yuyv" => VideoEncodeType::Yuyv,
            // Lets check if it's a valid codec string, otherwise will get unknown
            codec => VideoEncodeType::from_codec(codec),
        }
    }
}
```

Reply:
That sounds okay, but maybe:

```Rust
use serde::{Deserialize, Serialize};
use std::{fmt::Display, str::FromStr};

#[derive(Serialize, Deserialize, Eq, PartialEq, PartialOrd, Ord, Clone, Debug)]
#[serde(rename_all = "lowercase")]
pub enum VideoEncodeType {
    H264,
    H265,
    Mjpg,
    Unknown(String),
    Yuyv,
}

impl FromStr for VideoEncodeType {
    type Err = ();

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        let s = s.to_lowercase();
        match s.as_str() {
            "h264" => Ok(VideoEncodeType::H264),
            "h265" => Ok(VideoEncodeType::H265),
            "mjpg" => Ok(VideoEncodeType::Mjpg),
            "yuyv" => Ok(VideoEncodeType::Yuyv),
            _ => Ok(VideoEncodeType::Unknown(s)),
        }
    }
}

impl Display for VideoEncodeType {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        match self {
            VideoEncodeType::H264 => write!(f, "h264"),
            VideoEncodeType::H265 => write!(f, "h265"),
            VideoEncodeType::Mjpg => write!(f, "mjpg"),
            VideoEncodeType::Yuyv => write!(f, "yuyv"),
            VideoEncodeType::Unknown(s) => write!(f, "{}", s.to_lowercase()),
        }
    }
}

fn example() {
    assert!(VideoEncodeType::H264.to_string() == "h264");
    assert!(VideoEncodeType::from_str("h264").unwrap() == VideoEncodeType::H264);

    assert!(VideoEncodeType::Unknown("potato".into()).to_string() == "potato");
    assert!(
        VideoEncodeType::from_str("potato").unwrap()
            == VideoEncodeType::Unknown("potato".into())
    );
}
```